miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java

```diff
@@ -820,7 +828,13 @@ public String processSubmit(@ModelAttribute("sample") Sample sample, ModelMap mo
       if (builder.getTissueClass() != null) {
         builder.setTissueClass(sampleClassService.get(builder.getTissueClass().getId()));
       }
-      sample = builder.build();
+      sample = builder.build(new InferStockFromAliquot() {
+
+        @Override
+        public SampleClass infer(SampleClass sampleClass) {
+          return sampleClassService.inferStockFromAliquot(sampleClass);
```

Comment:
If there's no other reason to use this, SampleClassService could just have a method that returns an inferencer, so that a new one that does the same thing isn't created for every request. Or SampleClassService could just implement the interface itself